lib/services/identity.go

```diff
@@ -246,6 +248,54 @@ type AppSession interface {
 	DeleteAllAppSessions(context.Context) error
 }
 
+// WebTokensGetter provides access to web tokens
+type WebTokensGetter interface {
+	// WebTokens returns the tokens manager
+	WebTokens() WebTokenInterface
+}
+
+// WebTokenInterface defines interface for managing web tokens
+type WebTokenInterface interface {
+	// Get returns a token specified by the request.
+	Get(ctx context.Context, req types.GetWebTokenRequest) (types.WebToken, error)
```

Comment:
IMHO the generic names of the methods in this interface and `WebSessionInterface` below make it unfriendly to embedding since the names will clash. All other interfaces I think call them like `GetWebToken`, `DeleteAllWebTokens`, etc. Have you considered doing the same?

I see you have respective `XXXGetter` interfaces likely for this reason, maybe they also won't be needed if you just named the methods with "full" names? Unless I'm missing something.

Reply:
I still have a case when I need to support a couple of legacy APIs and I honestly did not want to come up with yet another prefixed or suffixed alternatives which required a different set of parameters. Also, I feel strongly against the flat namespace and I liked the direction that the kubernetes client took with its [Clientset](https://github.com/kubernetes/client-go/blob/c7a570ddfedd593495d663befa0f94bd127eac66/kubernetes/clientset.go#L70) and, for example, [CoreV1](https://github.com/kubernetes/client-go/blob/c7a570ddfedd593495d663befa0f94bd127eac66/kubernetes/typed/core/v1/core_client.go#L27) hence I made a small step towards the same thing here - using wrapper values with a well-defined set of APIs for all resources and differentiating them at the API entry with a corresponding getter.

The original layout is also unfriendly to versioning and this will eventually surface as an issue we'll need to solve.